src/main/java/com/powsybl/openloadflow/sa/WoodburyDcSecurityAnalysis.java

```diff
@@ -0,0 +1,272 @@
+
+            // set pre contingency angle states as state vector of equation system
+            context.getEquationSystem().getStateVector().set(preContingencyStates);
+
+            // Update network voltages with pre contingency states
+            updateNetwork(lfNetwork, context.getEquationSystem(), preContingencyStates);
+            if (context.getParameters().isSetVToNan()) {
+                for (LfBus bus : lfNetwork.getBuses()) {
+                    bus.setV(Double.NaN);
+                }
+            }
+
+            // update network result
+            var preContingencyNetworkResult = new PreContingencyNetworkResult(lfNetwork, monitorIndex, createResultExtension);
+            preContingencyNetworkResult.update();
+
+            // detect violations
+            var preContingencyLimitViolationManager = new LimitViolationManager(limitReductions);
+            preContingencyLimitViolationManager.detectViolations(lfNetwork);
+
+            // compute states with +1 -1 to model the contingencies and run connectivity analysis
+            ConnectivityBreakAnalysis.ConnectivityBreakAnalysisResults connectivityBreakAnalysisResults = ConnectivityBreakAnalysis.run(context, propagatedContingencies);
+
+            // save base state for later restoration after each contingency
+            NetworkState networkState = NetworkState.save(lfNetwork);
+
+            List<PostContingencyResult> postContingencyResults = new ArrayList<>();
+            LOGGER.info("Processing post contingency results for contingencies with no connectivity break");
+            connectivityBreakAnalysisResults.nonBreakingConnectivityContingencies()
+                    .forEach(contingency -> {
+                        // only process contingencies that impact the network
+                        if (!contingency.hasNoImpact()) {
+                            double[] postContingencyStates = calculatePostContingencyStatesForAContingency(context, connectivityBreakAnalysisResults.contingenciesStates(), preContingencyStates, contingency,
+                                    connectivityBreakAnalysisResults.contingencyElementByBranch(), Collections.emptySet(), Collections.emptySet(), reportNode, Collections.emptySet());
+                            PostContingencyResult postContingencyResult = computePostContingencyResult(context, securityAnalysisParameters, preContingencyLimitViolationManager,
+                                    preContingencyNetworkResult, contingency, postContingencyStates, limitReductions, createResultExtension);
+                            postContingencyResults.add(postContingencyResult);
+                            networkState.restore();
+                        }
+                    });
+
+            LOGGER.info("Processing post contingency results for contingencies breaking connectivity");
+            connectivityBreakAnalysisResults.connectivityAnalysisResults()
```

Comment:
does connectivityAnalysisResult return a connectivity result for the contingncis that break connectivity ?
If yes, the naming is not so great.